returnn/hdf.py

```diff
@@ -281,3 +283,69 @@ def run(self):
             audio.close()
 
         out_hdf.close()
+
+
+class RasrAlignmentDumpHDFJob(Job):
+    """
+    This Job reads Rasr alignment caches and dump them in hdf files.
+    """
+
+    def __init__(
+        self,
+        alignment_caches: List[tk.Path],
+        allophone_file: tk.Path,
+        state_tying_file: tk.Path,
+        data_type: type = np.uint16,
+        returnn_root: Optional[tk.Path] = None,
+    ):
+        """
+        :param alignment_caches: e.g. output of an AlignmentJob
+        :param allophone_file: e.g. output of a StoreAllophonesJob
+        :param state_tying_file: e.g. output of a DumpStateTyingJob
+        :param data_type: type that is used to store the data
+        :param returnn_root: file path to the RETURNN repository root folder
+        """
+        self.alignment_caches = alignment_caches
+        self.allophone_file = allophone_file
+        self.state_tying_file = state_tying_file
+        self.out_hdf_files = [self.output_path(f"data.hdf.{d}", cached=False) for d in range(len(alignment_caches))]
+        self.returnn_root = returnn_root
+        self.data_type = data_type
+        self.rqmt = {"cpu": 1, "mem": 8, "time": 0.5}
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt, args=range(1, (len(self.alignment_caches) + 1)))
+
+    def run(self, task_id):
+        state_tying = dict(
+            (k, int(v)) for l in open(self.state_tying_file.get_path()) for k, v in [l.strip().split()[0:2]]
+        )
+
+        alignment_cache = FileArchive(self.alignment_caches[task_id - 1].get_path())
```

Comment:
This assumes that the keys are 0,...,N. In our pipeline, they often start from 1, so it would be nice if that'd be possible here. In any case, it should be detected at graph creation time and not only fail when the job runs.

Reply:
self.alignment is a List[Path], those always start at 0.

Reply:
Oh, a list, right. I directly passed the `hidden_paths` of a `FlagDependentFlowAttribute`. But then I'll change it on my end.